stack/framework/hal/chips/stm32_common/stm32_common_timer.c

```diff
@@ -183,6 +183,7 @@ error_t hw_timer_schedule(hwtimer_id_t timer_id, hwtimer_tick_t tick )
  		return EOFF;
   //don't enable the interrupt while waiting to write a new compare value
   ready_for_trigger = false;
+  assert(!in_atomic()); // if we currently are in atomic context it is possible to have an infinite loop on the instruction below.
```

Comment:
now you are disabling the timer as a whole for something that could happen. At least also first check if cmp_reg_write_pending is also true